doc/src/guides/solving/solve-equation-algebraically.md

```diff
@@ -0,0 +1,279 @@
+# Solve an equation algebraically
+
+Use SymPy to solve an equation algebraically (symbolically). For example, solving $x^2 = y$ yields $x \in \{-\sqrt{y},\sqrt{y}\}$.
+
+There are two high-level functions to solve equations, {func}`~.solve` and {func}`~.solveset`. Here are recommendations on when to use:
+
+- {func}`~.solve`
+    - You need to programmatically extract components (expressions, or individual symbols or constants) from the output by traversing it.
+    - You want an explicit solution expression that you can use to substitute into something else.
+
+- {func}`~.solveset`
+    - You want a consistent input and output interface.
+    - You want to get all the solutions, including if there are infinitely many.
+    - You want a clear separation between equations in the complex domain and the real domain.
```

Comment:
Note that solve can also be used to find real solutions:
```python
In [1]: x = symbols('x')

In [2]: solve(x**4 - 1, x)
Out[2]: [-1, 1, -ⅈ, ⅈ]

In [3]: x = symbols('x', real=True)

In [4]: solve(x**4 - 1, x)
Out[4]: [-1, 1]
```

Reply:
Oh, I see you have that example below. What exactly does this statement about a "clear separation" mean?

Reply:
Clarified how solveset limits the domain of results.